sql/src/main/java/io/crate/lucene/CIDRRangeQuery.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.lucene;
+
+import io.crate.expression.operator.CIDROperator;
+import io.crate.expression.symbol.Function;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.lucene.search.Queries;
+import org.elasticsearch.index.mapper.MappedFieldType;
+
+import java.net.InetAddress;
+
+class CIDRRangeQuery implements FunctionToQuery {
+
+    @Override
+    public Query apply(Function input, LuceneQueryBuilder.Context context) {
+        RefAndLiteral refAndLiteral = RefAndLiteral.of(input);
+        if (refAndLiteral == null) {
+            return null;
+        }
+        String colName = refAndLiteral.reference().column().fqn();
+        MappedFieldType fieldType = context.getFieldTypeOrNull(colName);
+        if (fieldType == null) {
+            return Queries.newMatchNoDocsQuery("column does not exist in this index");
+        }
+        String cidr = (String) refAndLiteral.literal().value();
+        Tuple<InetAddress, InetAddress> cidrRange = CIDROperator.extractRange(cidr);
+        return fieldType.rangeQuery(cidrRange.v1(),
+                                    cidrRange.v2(),
+                                    true,
+                                    false,
+                                    null,
+                                    null,
+                                    context.queryShardContext);
+    }
```

Comment:
In `IpFieldMapper` the term query has the following logic:

```
                if (term.contains("/")) {
                    final Tuple<InetAddress, Integer> cidr = InetAddresses.parseCidr(term);
                    return InetAddressPoint.newPrefixQuery(name(), cidr.v1(), cidr.v2());
                }
```

Can we use that instead here as well?

Would probably also remove the need for `CIDROperator.extractRange(cidr);` and get rid of the `IpRangeConsumer` abstraction inside the `CIDROperator`

Reply:
InetAddressPoint.newPrefixQuery results in logic appropriate for `<<=`, however with `<<` the final IP of the cidr should not be not included (if we follow the postgresql doc). So I need take one out of the end address and perform a range query, which requires exposure of the CIDROperator.extractRange, which requires code duplication unless I use the Consumer.

Reply:
Do you have an example of a case that would match but shouldn't ?

I thought the `equals` portion of  `<<=` was to have `CIDR <<= CIDR` matching. But for `IP << CIDR` it shouldn't make a difference.